release/run_release_test.sh

```diff
@@ -67,8 +67,25 @@ fi
 if [ -z "${NO_CLONE}" ]; then
   TMPDIR=$(mktemp -d -t release-XXXXXXXXXX)
   echo "Cloning test repo ${RAY_TEST_REPO} branch ${RAY_TEST_BRANCH}"
-  git clone --depth 1 -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" "${TMPDIR}"
+  git clone -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" "${TMPDIR}"
   pushd "${TMPDIR}/release" || true
+  HEAD_COMMIT=$(git rev-parse HEAD)
+  echo "The cloned test repo has head commit of ${HEAD_COMMIT}"
+
+  # We only do this if RAY_TEST_REPO and RAY_TEST_BRANCH are pointing to ray master.
+  # Theoretically, release manager may also run into this issue when manually triggering
+  # release test runs. But cherry-picks are rare and thus it's less likely to run into
+  # this racing condition, ignoring for now.
+  if [ "${RAY_TEST_REPO}" == "https://github.com/ray-project/ray.git" ] && \
+  [[ "${PARSED_RAY_WHEELS}" == *"master"*  ]] && \
+  [ "${RAY_TEST_BRANCH}" == "master" ] && [ "${BUILDKITE_COMMIT}" ] && \
+  [ "${HEAD_COMMIT}" != "${BUILDKITE_COMMIT}" ]; then
+    echo "The checked out test code doesn't match with the installed wheel. \
+This is likely due to a racing condition when a PR is landed between \
+a wheel is installed and test code is checked out."
+    echo "Hard resetting from ${HEAD_COMMIT} to ${BUILDKITE_COMMIT}."
+    git reset --hard "${BUILDKITE_COMMIT}"
```

Comment:
Should we attempt to get the commit from the wheel though. BUILDKITE_COMMIT can be different from the wheel commit.

Reply:
I think the main use case is probably bisect, where each step tests different wheel commit. BUILDKITE_COMMIT is a constant across steps though and buildkite does not allow to change that.

Reply:
Ah I see. I didn't look much into the bisect tool. 
I changed the commit parsing logic into `get_step`. We will pass in a `ray-commit` arg into `run_release_test.sh`.

Reply:
I did that in `get_step` mainly because one feedback from Kai was to have `run_release_test.sh` being dumb.